esphome/components/tsl2591/tsl2591.h

```diff
@@ -0,0 +1,205 @@
+   * The disadvantage of powering down the device between readings
+   * is that you have to wait for the ADC to go through an
+   * integration cycle before a reliable reading is available.
+   * This happens during ESPHome's update loop, so waiting slows
+   * down the entire ESP device. You should only enable this if
+   * you need to minimize power consumption and you can tolerate
+   * that delay. Otherwise, keep the default of disabling
+   * power save mode.
+   *
+   * @param enable Enable or disable power save mode.
+   */
+  void set_power_save_mode(bool enable);
+
+  /** Sets the name for this instance of the device.
+   *
+   * @param name The user-friendly name.
+   */
+  void set_name(const char *name);
+
+  /** Calculates and returns a lux value based on the ADC readings.
+   *
+   * @param full_spectrum The ADC reading for the full spectrum sensor.
+   * @param infrared The ADC reading for the infrared sensor.
+   */
+  float get_calculated_lux(uint16_t full_spectrum, uint16_t infrared);
+
+  /** Get the combined illuminance value.
+   *
+   * This is encoded into a 32 bit value. The high 16 bits are the value of the
+   * infrared sensor. The low 16 bits are the sum of the combined sensor values.
+   */
+  uint32_t get_combined_illuminance();
+
+  /** Get an individual sensor channel reading.
+   *
+   * This gets an individual light sensor reading. Since it goes through
+   * the entire component read cycle to get one value, it's not optimal if
+   * you want to get all possible channel values. If you want that, first
+   * call `getCombinedIlluminance()` and pass that value to the companion
+   * method with a different signature.
+   *
+   * @param channel The sensor channel of interest.
+   */
+  uint16_t get_illuminance(TSL2591SensorChannel channel);
+
+  /** Get an individual sensor channel reading from combined illuminance.
+   *
+   * This gets an individual light sensor reading from a combined illuminance
+   * value, which you would obtain from calling `getCombinedIlluminance()`.
+   * This method does not communicate with the sensor at all. It's strictly
+   * local calculations, so it is efficient if you call it multiple times.
+   *
+   * @param channel The sensor channel of interest.
+   * @param combined_illuminance The previously obtained combined illuminance value.
+   */
+  uint16_t get_illuminance(TSL2591SensorChannel channel, uint32_t combined_illuminance);
+
+  /** Are the device ADC values valid?
+   *
+   * Useful for scripting. This should be checked before calling update().
+   * It asks the TSL2591 if the ADC has completed an integration cycle
+   * and has reliable values in the device registers. If you call update()
+   * before the ADC values are valid, you may cause a general delay in
+   * the ESPHome update loop.
+   *
+   * It should take no more than the configured integration time for
+   * the ADC values to become valid after the TSL2591 device is enabled.
+   */
+  boolean is_adc_valid();
```

Comment:
```suggestion
  bool is_adc_valid();
```
`boolean` is an Arduino-ism.

Reply:
what's an arduino? :-)